src/device.ts

```diff
@@ -396,6 +396,25 @@ export class Device extends vscode.Disposable {
         });
     }
 
+    public getSystemInfo(): Promise<string> {
+        return new Promise((resolve, reject) => {
+            this.exec('ev3dev-sysinfo').then(conn => {
```

Comment:
could use async here.

```
       return new Promise(async (resolve, reject) => {
            const conn = await this.exec('ev3dev-sysinfo');
            ...
```

Reply:
I'm not sure that is true. How would results be passed back from the event callbacks?

Reply:
By calling resolve()/reject().

See device.ts:412

Reply:
Oh, I see what you're saying; I misunderstood and thought you wanted to make the whole function async.